GUIDELINES.md

```diff
@@ -200,3 +227,41 @@ cluttering the output in the Travis-CI build environment.
 If an older version is required, put it in a subdirectory of the recipe.
 Examples of this can be found in [bowtie2](recipes/bowtie2),
 [bx-python](recipes/bx-python), and others.
+
+## ZLIB
+
+If a package depends on zlib, then it will most likely also depend on `gcc` (on
+Linux) and `llvm` (on OSX). The `meta.yaml` requirements section should
+therefore at least have the following:
+
+
+```yaml
+requirements:
+  build:
+    - gcc   # [not osx]
+    - llvm  # [osx]
+    - zlib
+
+  run:
+    - libgcc    # [not osx]
+    - zlib
+```
+
+When building the package, you may get an error saying that zlib.h can't be
+found -- despite having zlib listed in the dependencies. This is becuase the
+location of `zlib` often has to be specified in the `build.sh` script, which
+can be done like this:
+
+```bash
+export CFLAGS="-I$PREFIX/include"
+export LDFLAGS="-L$PREFIX/lib"
```

Comment:
Which is not needed? 

Seems like e.g. [verse](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/verse/build.sh) needed the CPATH;  [samtools](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/samtools/build.sh) and [cyvcf2](https://github.com/bioconda/bioconda-recipes/pull/1660/files) needed the CFLAGS and LDFLAGS. Since it's a difficult (for me at least) to resolve problem, I figured this was a good place to put the solution.

Is there a rule of thumb we could provide to suggest when this would or wouldn't be needed?